### PR TITLE
Feature/transactions

### PR DESCRIPTION
**Pull request by Joannis**
Implements a new transactions API by subclassing Collection and Database with an alternative that is explicitly used for transactions.

## Description

Also adapts some internal mechanisms:
- Adds a WireVersion to Cluster that is the minimum wire version from the last discovery
- Adds options for transactions as per spec (currently unused)
- Subclasses Collection and Database to create a temporary collection/database type specific to a transaction to carry context

## How Has This Been Tested?
- Tested it by making an example application as well as a unit test that tests both abort and commit
- Does not test bad use case flows (committing after abort)

## Checklist:
- [x] If applicable, I have updated the documentation accordingly.
- [x] If applicable, I have added tests to cover my changes.

### CODE REVIEW

File: Sources/MongoKitten/Commands/AggregateCommand.swift [hunk 1]
```diff
@@ -225,7 +227,7 @@ extension AggregateCursor where Element == Document {
             .getFirstResult()
             .thenThrowing { result in
                 guard let result = result else {
-                    throw MongoKittenError(.unexpectedNil, reason: .noResultDocument)
+                    return 0
```
Comment (Obbut) [inline Sources/MongoKitten/Commands/AggregateCommand.swift]: Why did you change this? Is there a case where MongoDB does not return a document here?

Comment (Joannis) [reply Sources/MongoKitten/Commands/AggregateCommand.swift]: No. MongoDB only returns a Document is the pipeline input >= 1 document. Which is ridiculous
